sphericart/include/cuda_base.hpp

```diff
@@ -66,29 +66,29 @@ namespace cuda {
  */
 template <typename scalar_t>
 void spherical_harmonics_cuda_base(
-    const scalar_t* __restrict__ xyz,
+    const scalar_t* xyz,
     const int nedges,
-    const scalar_t* __restrict__ prefactors,
```

Comment:
rather than removing __restrict__ maybe you can add an `#ifdef` clause that defines it to a noop when the compiler is not GCC